src/Database/Query.php

```diff
@@ -556,7 +554,7 @@ public function from($tables = [], $overwrite = false)
      * @param array $types associative array of type names used to bind values to query
      * @param bool $overwrite whether to reset joins with passed list or not
      * @see \Cake\Database\Type
-     * @return self
+     * @return $this
```

Comment:
Oh, yeah, indeed. We definitely need to fix that up soon, as well. Thanks, I adjusted the doc block.